modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java

```diff
@@ -588,6 +592,37 @@ public String toEnumName(CodegenProperty property) {
         }
     }
 
+    protected void setEnumPropertyNaming(String naming) {
+        try {
+            enumPropertyNaming = ENUM_PROPERTY_NAMING_TYPE.valueOf(naming);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Invalid enum property naming '" +
+                    naming + "'. Must be 'original', 'camelCase', " +
+                    "'PascalCase', 'snake_case', or 'UPPERCASE'");
```

Comment:
maybe the available values in `ENUM_PROPERTY_NAMING_TYPE` could be listed automatically, so the comment will not be outdated

Reply:
Good idea, will do!

Reply:
and done